app/controllers/sns_controller.rb

```diff
@@ -0,0 +1,53 @@
+# These endpoints allow the Amazon SES service to notify us when there are bounced
+# emails or complaints (ex we are marked as spam). The expected JSON is particular to
+# Amazon's SNS notification service, and thus will not work for other services.
+class SnsController < ApplicationController
+  protect_from_forgery with: :null_session
+  before_filter :verify_amazon_authorize_key
+
+  def bounce
+    begin
+      success = true
+      message = JSON.parse(request.body.read)['Message']
+      bounce_arr = JSON.parse(message)['bounce']['bouncedRecipients']
+      bounce_arr.each do |recipient|
+        Bounce.create(email: recipient['emailAddress'].downcase)
+      end
+    rescue
```

Comment:
Do you think it's reasonable to expect that the message from Amazon will conform to the formats they've described and let the code error if it doesn't? Or would it be better to check any possible errors based on unexpected format, ex. check that `message` isn't empty before the second call to `JSON.parse`?